can/interfaces/ics_neovi/neovi_bus.py

```diff
@@ -0,0 +1,255 @@
+        logger.info("Got configuration of: {}".format(config))
+
+        self._use_system_timestamp = bool(
+            config.get('use_system_timestamp', False)
+        )
+
+        try:
+            channel = int(channel)
+        except ValueError:
+            raise ValueError('channel must be an integer')
+
+        type_filter = config.get('type_filter')
+        serial = config.get('serial')
+        self.dev = self._open_device(type_filter, serial)
+
+        self.channel_info = '%s %s CH:%s' % (
+            self.dev.Name,
+            self.dev.SerialNumber,
+            channel
+        )
+        logger.info("Using device: {}".format(self.channel_info))
+
+        ics.load_default_settings(self.dev)
+
+        self.sw_filters = None
+        self.set_filters(can_filters)
+        self.rx_buffer = queue.Queue()
+        self.opened = True
+
+        self.network = int(channel) if channel is not None else None
+
+        # TODO: Change the scaling based on the device type
+        self.ts_scaling = (
+            ics.NEOVI6_VCAN_TIMESTAMP_1, ics.NEOVI6_VCAN_TIMESTAMP_2
+        )
+
+    def shutdown(self):
+        super(NeoViBus, self).shutdown()
+        self.opened = False
+        ics.close_device(self.dev)
+
+    def _open_device(self, type_filter=None, serial=None):
+        if type_filter is not None:
+            devices = ics.find_devices(type_filter)
+        else:
+            devices = ics.find_devices()
+
+        for device in devices:
+            if serial is None:
+                dev = device
+                break
+            if str(device.SerialNumber) == serial:
+                dev = device
+                break
+        else:
+            msg = ['No device']
+
+            if type_filter is not None:
+                msg.append('with type {}'.format(type_filter))
+            if serial is not None:
+                msg.append('with serial {}'.format(serial))
+            msg.append('found.')
+            raise Exception(' '.join(msg))
+        ics.open_device(dev)
+        return dev
+
+    def _process_msg_queue(self, timeout=None):
+        try:
+            messages, errors = ics.get_messages(self.dev, False, timeout)
+        except ics.RuntimeError:
+            return
+        for ics_msg in messages:
+            if ics_msg.NetworkID != self.network:
+                continue
+            if ics_msg.ArbIDOrHeader == 0:
+                # Looks like ICS device sends frames with ArbIDOrHeader = 0
+                # Need to find out exactly what they are for
+                # Filtering them for now
+                continue
+            if not self._is_filter_match(ics_msg.ArbIDOrHeader):
+                continue
+            self.rx_buffer.put(ics_msg)
+        if errors:
+            logger.warning("%d errors found" % errors)
+
+            for msg in ics.get_error_messages(self.dev):
+                logger.warning(msg)
+
+    def _is_filter_match(self, arb_id):
```

Comment:
I've removed the SW filtering from the other interfaces and created issue #157 to do a more general solution, so you could remove the filtering for now.

Reply:
When are you planing to add this feature in? We have code that currently depend on filtering the messages using this method.

Reply:
No plan. If you need it then definitely keep it. But it won’t work for all interfaces.